engine/access/ingestion2/results_forest_test.go

```diff
@@ -0,0 +1,1542 @@
+// when header retrieval fails.
+func (s *ResultsForestSuite) TestNewResultsForest() {
+	s.latestPersistedSealedResult.
+		On("Latest").
+		Return(s.initialSealedResult.ID(), s.initialSealedHeader.Height)
+
+	s.Run("successful creation", func() {
+		s.executionResults.
+			On("ByID", s.initialSealedResult.ID()).
+			Return(s.initialSealedResult, nil).
+			Once()
+
+		s.headers.
+			On("ByHeight", s.initialSealedHeader.Height).
+			Return(s.initialSealedHeader, nil).
+			Once()
+
+		pipeline := opsyncmock.NewPipeline(s.T())
+		pipeline.On("SetSealed").Return().Once()
+		s.pipelineFactory.
+			On("NewCompletedPipeline", s.initialSealedResult).
+			Return(pipeline).
+			Once()
+
+		forest, err := ingestion2.NewResultsForest(
+			s.log,
+			s.headers,
+			s.executionResults,
+			s.pipelineFactory,
+			s.latestPersistedSealedResult,
+			s.maxViewDelta,
+		)
+		s.Require().NoError(err)
+
+		s.NotNil(forest)
+		s.Equal(s.initialSealedHeader.View, forest.LowestView())
+	})
+
+	s.Run("result retrieval fails", func() {
+		s.executionResults.
+			On("ByID", s.initialSealedResult.ID()).
+			Return(nil, storage.ErrNotFound).
+			Once()
+
+		forest, err := ingestion2.NewResultsForest(
+			s.log,
+			s.headers,
+			s.executionResults,
+			s.pipelineFactory,
+			s.latestPersistedSealedResult,
+			s.maxViewDelta,
+		)
+		s.Error(err)
+		s.Nil(forest)
+	})
+
+	s.Run("header retrieval fails", func() {
+		s.executionResults.
+			On("ByID", s.initialSealedResult.ID()).
+			Return(s.initialSealedResult, nil).
+			Once()
+
+		s.headers.
+			On("ByHeight", s.initialSealedHeader.Height).
+			Return(nil, storage.ErrNotFound).
+			Once()
+
+		forest, err := ingestion2.NewResultsForest(
+			s.log,
+			s.headers,
+			s.executionResults,
+			s.pipelineFactory,
+			s.latestPersistedSealedResult,
+			s.maxViewDelta,
+		)
+		s.Error(err)
+		s.Nil(forest)
+	})
+}
+
+// TestResetLowestRejectedView tests the different scenarios of the ResetLowestRejectedView method,
+// and verifies that the correct values for the last sealed view and rejected status are returned.
+func (s *ResultsForestSuite) TestResetLowestRejectedView() {
+	resultGen := s.g.ExecutionResults()
+
+	s.Run("returns last sealed view and false when no results rejected", func() {
+		// start with the latest persisted sealed result already inserted
+		forest := s.createForest()
+
+		view, rejected := forest.ResetLowestRejectedView()
+
+		s.False(rejected)
+		s.Equal(s.initialSealedHeader.View, view)
+	})
+
+	s.Run("returns true when results have been rejected", func() {
+		// start with the latest persisted sealed result already inserted
+		forest := s.createForest()
+
+		// insert the first result that extends the latest persisted sealed result;
+		// this should never be rejected irrespective of its view, because it is the direct child of the
+		pipeline := s.createPipeline(s.results[0])
+		pipeline.On("SetSealed").Return().Once()
+
+		err := forest.AddSealedResult(s.results[0])
+		s.Require().NoError(err)
```

Comment:
suggestion: check what forest reports as its values for the latest sealed result (view and result)